### PR TITLE
plat-stm32mp1: clock: secure and non-secure gateable clocks

### PR DESCRIPTION
**Pull request by etienne-lms**
Array stm32mp1_clk_gate[] defines the clock resources. This change
adds an attribute to the clocks in stm32mp1_clk_gate array. Clocks
under RCC[TZEN] hardening are tagged SEC and clocks always assigned
to non-secure world as per SoC implementation are tagged N_S.

Non-secure clocks that OP-TEE expects to enable are enabled without
increase of their refcount and, for consistency, are never disabled
by TEE Core. Note that such clocks may be accessed by OP-TEE Core
when the non-secure world is not executing, for example at boot time
or could be when system is suspending/resuming.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

### CODE REVIEW

File: core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c [hunk 1]
```diff
@@ -213,46 +215,53 @@ struct stm32mp1_clk_pll {
 	enum stm32mp_osc_id refclk[REFCLK_SIZE];
 };
 
+#define N_S	0	/* Non-secure can access RCC interface */
+#define SEC	1	/* RCC[TZEN] protects RCC interface */
```
Comment (jforissier) [inline core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c:219]: Minor: how about `SEC` and `NSEC`?

Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c:219]: no strong opinion but I think `N_S`/`SEC` better differs versus `NSEC`/`SEC`.

Comment (jforissier) [reply core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c:219]: Fine with me

**Comment by etienne-lms** [comment]
tag applied

**Pull request merged**